diff --git a/include/libuvc/libuvc.h b/include/libuvc/libuvc.h
index d387150..099b2dd 100644
--- a/include/libuvc/libuvc.h
+++ b/include/libuvc/libuvc.h
@@ -73,6 +73,7 @@ enum uvc_frame_format {
   /** Motion-JPEG (or JPEG) encoded images */
   UVC_FRAME_FORMAT_MJPEG,
   UVC_FRAME_FORMAT_H264,
+  UVC_FRAME_FORMAT_H265,
   /** Greyscale images */
   UVC_FRAME_FORMAT_GRAY8,
   UVC_FRAME_FORMAT_GRAY16,
diff --git a/src/stream.c b/src/stream.c
index b7b88af..3f3752f 100644
--- a/src/stream.c
+++ b/src/stream.c
@@ -135,12 +135,14 @@ struct format_table_entry *_get_format_entry(enum uvc_frame_format format) {
       {'R',  'G',  'G',  'B', 0x00, 0x00, 0x10, 0x00, 0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71})
     FMT(UVC_FRAME_FORMAT_SBGGR8,
       {'B',  'G',  'G',  'R', 0x00, 0x00, 0x10, 0x00, 0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71})
-    ABS_FMT(UVC_FRAME_FORMAT_COMPRESSED, 2,
-      {UVC_FRAME_FORMAT_MJPEG, UVC_FRAME_FORMAT_H264})
+    ABS_FMT(UVC_FRAME_FORMAT_COMPRESSED, 3,
+      {UVC_FRAME_FORMAT_MJPEG, UVC_FRAME_FORMAT_H264, UVC_FRAME_FORMAT_H265})
     FMT(UVC_FRAME_FORMAT_MJPEG,
       {'M',  'J',  'P',  'G'})
     FMT(UVC_FRAME_FORMAT_H264,
       {'H',  '2',  '6',  '4', 0x00, 0x00, 0x10, 0x00, 0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71})
+    FMT(UVC_FRAME_FORMAT_H265,
+      {'H',  '2',  '6',  '5', 0x00, 0x00, 0x10, 0x00, 0x80, 0x00, 0x00, 0xaa, 0x00, 0x38, 0x9b, 0x71})
 
     default:
       return NULL;
@@ -1361,6 +1363,9 @@ void _uvc_populate_frame(uvc_stream_handle_t *strmh) {
   case UVC_FRAME_FORMAT_H264:
     frame->step = 0;
     break;
+  case UVC_FRAME_FORMAT_H265:
+    frame->step = 0;
+    break;
   default:
     frame->step = 0;
     break;
